sources/Algo/src/Algo.cpp

```diff
@@ -246,5 +249,277 @@ ControllerInterfaceDefinitionAlgorithm::operator()(const NodePtr& node) {
+
+DynModelAlgorithm::DynModelAlgorithm(DynModelDefinitions& models, const inputs::DynamicDataBaseManager& manager) : models_(models), manager_(manager) {
+  extractDynModels();
+}
+
+boost::optional<boost::filesystem::path>
+DynModelAlgorithm::computeLibPath(const std::string& lib) {
+  static auto dflLibEnv = getenv("DYNAFLOW_LAUNCHER_LIBRARIES");
+
+  auto libFile = lib + DYN::sharedLibraryExtension();
+  // Check dynawo path
+  auto libPath = DYN::getLibraryPathFromName(libFile);
+  if (libPath) {
+    return libPath;
+  }
+
+  if (dflLibEnv == NULL) {
+    return boost::none;
+  }
+
+  auto libPathDfl = boost::filesystem::path(dflLibEnv);
+  libPathDfl.append(libFile);
+  if (!boost::filesystem::exists(libPathDfl)) {
+    return boost::none;
+  }
+
+  return boost::make_optional(libPathDfl);
+}
+
+bool
+DynModelAlgorithm::doesLibraryExist(const std::string& lib) {
+  try {
+    // check DFL local path
+    auto libPath = computeLibPath(lib);
+    if (!libPath) {
+      return false;
+    }
+    boost::dll::shared_library sharedLib(libPath->generic_string());
+    static_cast<void>(sharedLib);  // do nothing as we just want to check that the library can be loaded
+    return true;
+  } catch (const std::exception& e) {
+    LOG(warn) << MESS(CannotLoadLibrary, lib, e.what()) << LOG_ENDL;
+    return false;
+  }
+}
+
+void
+DynModelAlgorithm::dispatchAutomatonMulti(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton,
+                                          const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                                          const inputs::AssemblingXmlDocument::MultipleAssociation& multiassoc) {
+  MacroConnect connection(automaton.id, macro.macroConnection);
+  const auto& vlid = multiassoc.shunt.voltageLevel;
+  macroConnectByVlForShuntsId_[vlid].push_back(connection);
+}
+
+void
+DynModelAlgorithm::dispatchAutomatonSingle(const inputs::AssemblingXmlDocument::DynamicAutomaton& automaton,
+                                           const inputs::AssemblingXmlDocument::MacroConnect& macro,
+                                           const inputs::AssemblingXmlDocument::SingleAssociation& singleassoc) {
+  MacroConnect connection(automaton.id, macro.macroConnection);
+  if (singleassoc.bus) {
+    macroConnectByVlForBusesId_[singleassoc.bus->voltageLevel].insert(connection);
+  } else if (singleassoc.line) {
+    macroConnectByLineName_[singleassoc.line->name].push_back(connection);
+  } else if (singleassoc.tfo) {
+    macroConnectByTfoName_[singleassoc.tfo->name].push_back(connection);
+  } else {
+// Cannot happen if xml file is formed correctly according to its XSD
+#if _DEBUG_
+    assert(false);
+#endif
+  }
+}
+
+void
+DynModelAlgorithm::extractDynModels() {
+  using inputs::AssemblingXmlDocument;
+
+  const auto& automatons = manager_.assemblingDocument().dynamicAutomatons();
+  const auto& singleassociations = manager_.assemblingDocument().singleAssociations();
```

Comment:
coding style singleAssociations (same for multi)

Reply:
OK